lib/Doctrine/DBAL/Cache/ResultCacheStatement.php

```diff
@@ -59,11 +61,12 @@ class ResultCacheStatement implements IteratorAggregate, ResultStatement
     private $defaultFetchMode = FetchMode::MIXED;
 
     /**
-     * @param string $cacheKey
-     * @param string $realKey
-     * @param int    $lifetime
+     * @param ResultStatement|PDOStatement $stmt
+     * @param                              string                       $cacheKey
+     * @param                              string                       $realKey
+     * @param                              int                          $lifetime
      */
-    public function __construct(ResultStatement $stmt, Cache $resultCache, $cacheKey, $realKey, $lifetime)
+    public function __construct($stmt, Cache $resultCache, $cacheKey, $realKey, $lifetime)
```

Comment:
Even when passing the PDO instance we modify the statement class, so I'd say we shouldn't remove this type hint here.

Reply:
PHPStan was complaining about it, that's why I had to modify it.